text/0685-new-browser-support-policy.md

```diff
@@ -0,0 +1,416 @@
+Ember Data.
+
+## Motivation
+
+With Microsoft's recent release of the new Chromium-based Edge browser, which
+has a compatibility mode for Internet Explorer built in, many frameworks, tools,
+libraries, and websites have begun finally dropping support for the aging
+browser. In order to unlock the latest browser features and continue improving
+the framework as a whole, Ember should also drop support in the next major
+release.
+
+In dropping support for Internet Explorer, we will need a new browser support
+policy. Until now Internet Explorer has been the "lowest common denominator"
+for all features. It is a very old browser that no longer releases new versions,
+and was not "evergreen" (e.g. constantly updating) even when it was. As such,
+Ember has been very stable from release to release, as supporting any version of
+Internet Explorer meant we simply could not use any new browser features.
+
+With modern evergreen browser release cycles, where browsers release regularly
+and users are generally on one of the more recent versions of a browser, this
+dynamic changes. We cannot set an explicit version to support anymore, because
+it would be impractical - Chrome alone releases a new version every 6 weeks.
+Setting an explicit lowest-supported-version would lock us into having to
+support a huge number of older browsers, many of which are entirely unused.
+
+Conversely, however, tracking the latest release of a browser may not be enough
+stability for Ember apps. If Ember were to adopt a new browser feature
+immediately after its release, even though many web users were still using a
+previous version, it would cause many Ember apps to either break, or be unable
+to update until their users had also updated to the latest browser. Even a
+rolling `N - 1` or `N - 2` support policy, where the last 1 or 2 versions are
+supported, may not be enough. Many users were stuck in the recent Edgium update
+for some time, for instance.
+
+This RFC seeks to establish a new support policy that addresses these issues
+with a new set of heuristics. These heuristics distinguish between two types of
+browsers which are supported:
+
+- **Evergreen browsers**, those which have adopted a continuous update policy
+  and generally keep their users up-to-date automatically.
+- **Non-evergreen browsers**, which have more complicated support patterns that
+  may prevent users from updating regularly.
+
+Depending on which category a browser falls into, different rules will be
+applied. This will allow us to effectively support browsers which release
+frequently, such as Chrome, and browsers which do not, such as Safari.
+
+## Proposed policy
+
+In the new support policy, Ember will support the following browsers:
+
+- Desktop
+  1. Google Chrome
+  2. Mozilla Firefox
+  3. Microsoft Edge
+  4. Safari
+- Mobile
+  1. Google Chrome
+  2. Mozilla Firefox
+  3. Safari
+- Testing
+  1. Headless Chrome
+  2. Headless Safari
+
```

Comment:
What is the reason to not support headless Firefox for testing?

Reply:
Sorry, this was a typo, it should be headless firefox not headless safari